src/openMVG/sfm/sfm_data_BA.hpp

```diff
@@ -7,20 +7,37 @@
 #ifndef OPENMVG_SFM_DATA_BA_HPP
 #define OPENMVG_SFM_DATA_BA_HPP
 
+#include <openMVG/stl/bitmask.hpp>
+
 namespace openMVG {
 namespace sfm {
 
+enum BA_Refine
+{
+  BA_REFINE_NONE = 0,
+  BA_REFINE_ROTATION = 1, //< refine pose rotations
+  BA_REFINE_TRANSLATION = 2, //< refine pose translations
+  BA_REFINE_INTRINSICS = 4, //< refine camera intrinsics
+  BA_REFINE_STRUCTURE = 8, //<  refine structure (i.e. 3D points)
+  /// Refine all parameters
+  BA_REFINE_ALL = BA_REFINE_ROTATION | BA_REFINE_TRANSLATION | BA_REFINE_INTRINSICS | BA_REFINE_STRUCTURE,
+};
+
+OPENMVG_BITMASK(BA_Refine)
+
```

Comment:
Boolean operators are explicit and standard.
IMHO, there is no reason to add more glue on it.